app/src/main/java/com/example/tema5/MainActivity.java

```diff
@@ -0,0 +1,111 @@
+package com.example.tema5;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+
+public class MainActivity extends AppCompatActivity {
+    private final String TAG = "MainActivity";
+    private String text;
+    private EditText websiteText, locationText, shareText, phoneText;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main_ex4);
+        websiteText = findViewById(R.id.website_text);
+        locationText = findViewById(R.id.location_text);
+        shareText = findViewById(R.id.share_text);
+        phoneText = findViewById(R.id.call_text);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        Log.e(TAG, new Throwable().getStackTrace()[0].getMethodName());
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        Log.e(TAG, new Throwable().getStackTrace()[0].getMethodName());
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        Log.e(TAG, new Throwable().getStackTrace()[0].getMethodName());
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();
+        Log.e(TAG, new Throwable().getStackTrace()[0].getMethodName());
```

Comment:
A nice approach to use reflection 👍

Reply:
You could write your own logger in here, to not copy-paste the same initialization in all methods.